api/v1/address.go

```diff
@@ -1,55 +1,104 @@
+func ConfigureAddressesRouter(e *echo.Group, service *db.SpdTenantSvc) {
+	g := e.Group("/addresses")
+	g.PUT("", handleSetAddresses)
+	g.DELETE("", handleDeleteAddresses)
+	g.GET("", handleGetAddresses)
 }
 
 // handleSetAddresses godoc
 //
-//			@Summary		Update addresses associated with a tenant
-//	 		@Security apiKey
-//		  @Param 			addresses body AddressesMutable true "New addresses to add or change is_signing flag of"
-//			@Produce		json
-//			@Success		200	{object}	ResponseEnvelope{response=Addresses}
-//			@Router			/addresses [put]
+//	@Summary		Update addresses associated with a tenant
+//	@Param			token header string true "Auth token"
+//	@Param			addresses body AddressesMutable true "New addresses to add or change is_signing flag of"
+//	@Produce		json
+//	@Success		200	{object}	ResponseEnvelope{response=Addresses}
+//	@Router			/addresses [put]
 func handleSetAddresses(c echo.Context) error {
-	return c.JSON(http.StatusNotImplemented, map[string]string{})
+	var addresses Addresses
+	err := json.NewDecoder(c.Request().Body).Decode(&addresses)
+
+	if err != nil {
+		return c.JSON(http.StatusInternalServerError, CreateErrorResponseEnvelope(c, http.StatusInternalServerError, err.Error()))
+	}
+
+	err = db.DB.Transaction(func(tx *gorm.DB) error {
+		for _, address := range addresses {
+			address.TenantID = GetTenantId(c)
+			res := tx.Where("tenant_id = ? AND address_robust in (?) AND deleted_at IS NULL").Save(&address)
+			if res.Error != nil {
+				return err
+			}
+		}
+
+		return nil
+	})
+
+	if err != nil {
+		return c.JSON(http.StatusInternalServerError, CreateErrorResponseEnvelope(c, http.StatusInternalServerError, err.Error()))
+	}
+
+	return c.JSON(http.StatusOK, CreateSuccessResponseEnvelope(c, "Updated Addresses associated with the tenant"))
 }
 
 // handleDeleteAddresses godoc
 //
-//		@Summary		Delete addresses used by a tenant
-//		@Security apiKey
-//	  @Param 			addresses body []string true "addresses to delete"
-//		@Produce		json
-//		@Success		200	{object}	ResponseEnvelope{response=Addresses}
-//		@Router			/addresses [delete]
+//	@Summary		Delete addresses used by a tenant
+//	@Param 			token header string true "Auth token"
+//	@Param 			addresses body []string true "addresses to delete"
+//	@Produce		json
+//	@Success		200	{object}	ResponseEnvelope{response=Addresses}
+//	@Router			/addresses [delete]
 func handleDeleteAddresses(c echo.Context) error {
-	return c.JSON(http.StatusNotImplemented, map[string]string{})
+	var addressesIds []string
+	err := json.NewDecoder(c.Request().Body).Decode(&addressesIds)
+
+	if err != nil {
+		return c.JSON(http.StatusInternalServerError, CreateErrorResponseEnvelope(c, http.StatusInternalServerError, err.Error()))
+	}
+
+	err = db.DB.Where("tenant_id = ? AND address_robust in (?) AND deleted_at IS NULL", GetTenantId(c), addressesIds).Delete(&Address{}).Error
```

Comment:
more gorm idiomatic way to do it is like

```
	toDelete := Address{TenantID: GetTenantId(c), AddressRobust: addressesIds[0]}
	db.DB.Delete(&toDelete)
```

Reply:
you should even be able to do in a batch by making `toDelete` an array of `[]Address` and then passing that in instead

Reply:
Issue seems to be the tenant id isnt persisted on the object outside of a loop. Tried doing it with create and it didnt work